v1/ratelimit/redis_delayed_sync_pipelined.go

```diff
@@ -336,7 +340,23 @@ func (r *RedisDelayedSyncPipelined) processSyncRes(cmdArgs syncArgs, cmdRes inte
 	case AdjustLocal:
 		diff := vals[1].(int64)
 		remote := vals[2].(int64)
-		cmdArgs.lmt.IncrementResetAtBy(diff)
+
+		if ls, ok := cmdArgs.lastSynced.(int64); !ok || ls == 0 {
+			// This is the "drift" case where this server has synced for the first time with Redis
+			// In this case, Redis will ask the local server to adjust its resetAt to be in sync with
+			// the global state and diff will be equal to remoteResetAt - cmdArgs.resetAt
+			// We need to account for the fact that the local resetAt might have changed, and thus
+			// adjust based on the current resetAt.
+			// A better solution is to modify the Lua script to explicitly handle this case, but it would
+			// break compatibility for existing clients and require them to manually delete the script from the Redis
+			// server
+			diff += cmdArgs.resetAt - cmdArgs.lmt.GetResetAt()
```

Comment:
I think you gotta minus the `delta` away. 
```suggestion
			diff += cmdArgs.resetAt - (cmdArgs.lmt.GetResetAt() - cmdArgs.lmt.GetDelta())
```

Ideally, `cmdArgs.lmt.GetResetAt() - cmdArgs.lmt.GetDelta()` can be derived atomically

But actually `cmdArgs.lmt.GetResetAt() - cmdArgs.lmt.GetDelta()` is just `now() - burst` before the sync

Reply:
Why do we need to include the local delta? 

`diff = remoteResetAt - argResetAt`   but we're trying to get ` diff = removeResetAt - currentLocalResetAt` 

I don't understand where `delta` comes into play